src/mon/AuthMonitor.cc

```diff
@@ -1929,6 +1876,145 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
   return false;
 }
 
+/* Generate the caps that should be present in the entity's auth keyring
+ * after running the "fs authorize" command. This is done by merging the
+ * caps already present in the client's auth keyring with the new caps
+ * provided by the user at "fs authorize" command.
+ */
+bool AuthMonitor::_gen_wanted_caps(EntityAuth& e_auth,
+  vector<string>& newcaps, ostream* out)
+{
+  bool all_caps_same = true; // whether all caps passed are same as the
+			     // one present.
+
+  if (e_auth.caps.empty()) {
+    all_caps_same = false;
+    return all_caps_same;
+  }
+
+  for (auto it = newcaps.begin(); it != newcaps.end(); it += 2) {
+    string& cap_entity = *it;
+    string& n_cap_str = *(it + 1); // new cap to be added to the current cap
```

Comment:
That's not possible because the vector is created in the same patch to have 6 elements.

Reply:
IMO adding assert is redundant because it has been hardcoded to have 6 members. Copying from rishabh-d-dave:ceph src/mon/AuthMonitor.cc
```
    vector<string> newcaps = {"mon", mon_cap_string, "osd", osd_cap_string,
			      "mds", mds_cap_string};
```

Reply:
In case this is used in code elsewhere. Also, what about comment: https://github.com/ceph/ceph/pull/41779/commits/e2f4e203a0c9b7687bf7577f8ed7989b2ff7c1b9#r885461143